### ISSUE TITLE
"key contains no private key" could really mean "EC private keys not supported"

### ISSUE THREAD

**Issue by mac-chaffee** [joseluisq/static-web-server#207]
**Is your feature request related to a problem? Please describe.**
When using `http2 = true` with TLS alongside the [Lego ACME client](https://go-acme.github.io/lego/usage/cli/options/) which defaults to generating EC256 keys, SWS fails to start with the following error:

```
2023-05-28T03:45:38.129495Z ERROR static_web_server::server: server failed to start up: failed to initialize TLS probably because invalid cert or key file
Caused by:
     key contains no private key
```

This error goes away when I force Lego to give me an rsa4096 key instead. Seems like the [Warp code](https://github.com/static-web-server/static-web-server/blob/c64e3d6754fc35974a0d0872b311d06505ae568a/src/tls.rs#L211) doesn't support Elliptic Curve keys, which begin with `-----BEGIN EC PRIVATE KEY-----`.

**Describe the solution you'd like**
Seems like [rustls_pemfile](https://docs.rs/rustls-pemfile/latest/rustls_pemfile/) crate does support EC keys if you call `read_one` and do pattern matching on the result. Not sure if the rest of the code will accept it.

**Describe alternatives you've considered**
Generate RSA keys instead. Or just make the error message clearer if EC keys aren't easy to support ATM.

**Additional context**
N/A

**Comment by joseluisq** [comment]
Good catch!
You're right. The warp code does not support EC private keys but `rustls_pemfile` does.
However, giving this a quick look up and trying locally. I found that effectively, it works for SWS by doing the proper pattern matching for `rustls_pemfile::Item::ECKey(key)`.

So a feature PR should come next to support EC private keys in the minor next release.

**Comment by joseluisq** [comment]
BTW this is probably related to #148.

**Comment by mac-chaffee** [comment]
That was fast, thanks!

Also found this as another workaround for people on older versions: https://github.com/seanmonstar/warp/issues/1014#issuecomment-1366412822

**Comment by joseluisq** [comment]
Ready for next minor. So a test from your side in the meantime should be appreciated.

**Comment by mac-chaffee** [comment]
My test was successful! Thanks!

**Comment by joseluisq** [comment]
Released on [v2.17.0](https://github.com/static-web-server/static-web-server/releases/tag/v2.17.0)
